### PR TITLE
refactor: port div_ceil from stdlib to avoid unstable features

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

When using our code base with those repos requires stable toolchains, like vector, unstable features block its compilation. This patch removes replaceable unstable feature in time module.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/common/time/src/timestamp.rs [hunk 1]
```diff
@@ -143,7 +143,8 @@ impl Timestamp {
             Some(Timestamp::new(value, unit))
         } else {
             let mul = unit.factor() / self.unit().factor();
-            Some(Timestamp::new(self.value.div_ceil(mul as i64), unit))
+            let new_ts = self.value as f64 / mul as f64;
```
Comment (evenyag) [inline src/common/time/src/timestamp.rs:146]: I think we should port the implementation from https://github.com/rust-lang/rust/pull/88582

Comment (v0y4g3r) [inline src/common/time/src/timestamp.rs:146]: I'm afraid this conversion will lose some precision. For example, let `mul` = 1000, `self.value` = 9223372036854676001 (a random number big enough),  `self.value.div_ceil(mul)` gives you 9223372036854677 while `(self.value as f64 / mul as f64).ceil() as i64` gives you 9223372036854676, which is incorrect.

Comment (v0y4g3r) [reply src/common/time/src/timestamp.rs:146]: Maybe we can port the `div_ceil` function:
```rs
#[inline]
pub const fn div_ceil(self, rhs: Self) -> Self {
    let d = self / rhs;
    let r = self % rhs;
    if (r > 0 && rhs > 0) || (r < 0 && rhs < 0) {
        d + 1
    } else {
        d
    }
}
```

Comment (sunng87) [reply src/common/time/src/timestamp.rs]: Thanks for searching for this.

Comment (sunng87) [reply src/common/time/src/timestamp.rs]: Confirmed. It happens when `self.value` is big enough for 64-bit value.

**Comment by sunng87** [comment]
Just ported rust stdlib's impl of `div_ceil` and add reproducible test case from @v0y4g3r

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**